src/dma/mod.rs

```diff
@@ -1080,8 +1112,15 @@ where
     /// * The closure `f` takes too long to return and a buffer overrun occurs.
     ///
     /// # Safety
-    /// Memory corruption might occur if the previous buffer, the one passed to the closure, gets
-    /// dropped in the closure and an overrun occurs in double buffering mode.
+    ///
+    /// Memory corruption might occur in the previous buffer, the one passed to the closure, if an
+    /// overrun occurs in double buffering mode. This method will panic if it detects an overrun,
+    /// however, the overrun won't be detected if the closure takes long enough to cause the DMA to
+    /// change buffers twice since the call of this method.
```

Comment:
Shouldn't we be able to know if DMA changed the buffers twice by looking at the interrupt flags? I guess we'd have to clear them before invoking the closure and I'm not sure if we can/should. But if we can, we can just check those flags afterwards and if one is set, that means an overrun occurred.

Reply:
You are right, we should be able to detect that by checking the flags, we might also do that instead of reading the address register at the two methods, I will implement this. Also, we are already clearing the flag at the beginning of the method and I should document it.